client/my-sites/email/form/mailboxes/validators.ts

```diff
@@ -255,40 +255,63 @@ class PasswordValidator extends BaseValidator< string > {
 
 class ExistingMailboxNamesValidator extends BaseValidator< string > {
 	private readonly existingMailboxNames: string[];
-	private readonly domainName: string;
+	protected readonly domainName: string;
 
 	constructor( domainName: string, existingMailboxNames: string[] ) {
 		super();
 		this.domainName = domainName;
 		this.existingMailboxNames = existingMailboxNames;
 	}
 
-	static getExistingMailboxError( domainName: string, existingMailbox: string ): FieldError {
+	static getExistingMailboxError( domainName: string, existingMailboxName: string ): FieldError {
 		return i18n.translate(
 			'Please use unique email addresses. {{strong}}%(emailAddress)s{{/strong}} already exists in your account.',
 			{
-				args: { emailAddress: `${ existingMailbox }@${ domainName }` },
+				args: { emailAddress: `${ existingMailboxName }@${ domainName }` },
 				components: { strong: createElement( 'strong' ) },
 			}
 		);
 	}
 
+	getMailboxError( fieldValue: string ) {
+		return ExistingMailboxNamesValidator.getExistingMailboxError( this.domainName, fieldValue );
+	}
+
 	validateField( field: MailboxFormFieldBase< string > ): void {
-		const existingMailboxNames = this.existingMailboxNames ?? [];
-		if ( ! existingMailboxNames || ! field.value ) {
+		if ( ! field.value || field.error ) {
 			return;
 		}
 
-		const fieldValueLowerCased = field.value.toLowerCase();
+		const fieldValueLowerCased = field.value.trim().toLowerCase();
```

Comment:
Why do call `.trim()` here?

Reply:
I'm mitigating for scenarios where users may copy and paste text with trailing or leading space.

Reply:
Wouldn't `sanitizeMailboxValue` already have the same effect..? Looks like it strips out all whitespace characters AFAICT.

Reply:
It does that and more, and it works okay when you derive, and suggest the mailbox name from the value of another field.

However we shouldn't strip intentional characters during the validation stage. We should only compare the input supplied by the user, except for leading or trailing space IMO.